File: apps/web/src/code/js/timeout/timeout_filesystem.js
import { Sandbox } from 'e2b'

const sandbox = await Sandbox.create({ template: 'base' })

// Timeout 3s for the write operation
await sandbox.filesystem.write('hello.txt', 'Hello World!', {timeout: 3000}) // $HighlightLine

// Timeout 3s for the list operation
const files = await sandbox.filesystem.list('.', {timeout: 3000}) // $HighlightLine
console.log(files)

// Timeout 3s for the read operation
const content = await sandbox.filesystem.read('hello.txt', {timeout: 3000}) // $HighlightLine
console.log(content)

await sandbox.close()

File: apps/web/src/code/python/timeout/timeout_filesystem.py
from e2b import Sandbox

sandbox = Sandbox(template="base")

# Timeout 3s for the write operation
sandbox.filesystem.write("test.txt", "Hello World", timeout=3)  # $HighlightLine

# Timeout 3s for the list operation
dir_content = sandbox.filesystem.list(".", timeout=3)  # $HighlightLine
print(dir_content)

# Timeout 3s for the read operation
file_content = sandbox.filesystem.read("test.txt", timeout=3)  # $HighlightLine
print(file_content)

sandbox.close()
